src/item.cpp

```diff
@@ -197,25 +197,15 @@ item::item(JsonObject &jo)
     deserialize(jo);
 }
 
-item::~item()
-{
-}
-
-void item::make( const std::string new_type, bool scrub )
+item& item::convert( const itype_id& new_type )
 {
     type = find_type( new_type );
-    contents.clear();
-
-    if (scrub) {
-        components.clear();
-        charges = -1;
-        bday = 0;
-        name = "";
-        curammo = NULL;
+    if( is_null() ) {
```

Comment:
> Somewhat relatedly has the use of exceptions ever been discussed?

Pretty sure the result was "they're not worth the trouble". They're used for json reader, but that's a very specific case.
As far as I know, C++ doesn't have strict exception handling enforcement rules like Java, so adding them to more general functions would require very careful handling.